packages/nx/src/command-line/release/version.ts

```diff
@@ -610,3 +633,43 @@ function resolveGeneratorData({
     throw err;
   }
 }
+function runPreVersionCommand(
+  preVersionCommand: string,
+  { dryRun, verbose }: { dryRun: boolean; verbose: boolean }
+) {
+  if (!preVersionCommand) {
+    return;
+  }
+
+  output.logSingleLine(`Executing pre-version command`);
+  if (verbose) {
+    console.log(`Executing the following pre-version command:`);
+    console.log(preVersionCommand);
+  }
+
+  let env: Record<string, string> = {
+    ...process.env,
+  };
+  if (dryRun) {
+    env.NX_DRY_RUN = 'true';
+  }
+
+  const stdio = verbose ? 'inherit' : 'pipe';
+  try {
+    execSync(preVersionCommand, {
```

Comment:
Can we use `runCommand` from our `native` code here when not on CI? Its fine to use execSync if on CI, but locally we'll get better looking output if we use the pty.

Reply:
per outside discussion - I'm going to leave this as is for now. We could rework it for a slight improvement, but for now it's fine as is since it's behind a --verbose flag.